tiproxy/tiproxy-grafana.md

```diff
@@ -0,0 +1,61 @@
+---
+title: TiProxy Monitoring Metrics
+summary: Learn the monitoring items of TiProxy.
+---
+
+# TiProxy Monitoring Metrics
+
+This document describes the monitoring items of TiProxy.
+
+If you use TiUP to deploy the TiDB cluster, the monitoring system (Prometheus & Grafana) is deployed at the same time. For more information, see [Overview of the Monitoring Framework](/tidb-monitoring-framework.md).
+
+The Grafana dashboard is divided into a series of sub dashboards which include Overview, PD, TiDB, TiKV, TiProxy, and Node\_exporter. A lot of metrics are there to help you diagnose. Each dashboard contains panel groups and their panels.
+
+TiProxy has four panel groups. The metrics on these panels indicate the current status of TiProxy.
+
+- **TiProxy-Server**: instance information.
+- **TiProxy-Query-Summary**: SQL query metrics like CPS. 
+- **TiProxy-Backend**: information on TiDB nodes that TiProxy might connect to.
+- **TiProxy-Balance**: loadbalance mertrics.
+
+## Server
+
+- CPU Usage: the CPU utilization of each TiProxy instance
+- Memory Usage: the memory usage of each TiProxy instance
+- Uptime: the runtime of each TiProxy instance since last restart
+- Connection Count: the number of clients connected to each TiProxy instance
+- Create Connection OPM: the number of creating connections on each TiProxy instance every minute
+- Disconnection OPM: the number of disconnections for each reason every minute. Reasons include:
+  - success: the client disconnects normally
+  - client network break: the client does not send a `QUIT` command before it disconnects. It may also be caused by a network problem or the client shutting down
+  - client handshake fail: the client fails to handshake with TiProxy
+  - auth fail: the access is denied by TiDB
+  - SQL error: TiDB returns other SQL errors
+  - proxy shutdown: TiProxy is shutting down
+  - malformed packet: TiProxy fails to parse the MySQL packet
+  - get backend fail: TiProxy fails to find an available backend for the connection
+  - proxy error: other TiProxy errors
+  - backend network break: fails to read from or write to the TiDB. This may be caused by a network problem or the TiDB server shutting down
+  - backend handshake fail: TiProxy fails to handshake with the TiDB server
```

Comment:
```suggestion
    - success: the client disconnects normally
    - client network break: the client does not send a `QUIT` command before it disconnects. It may also be caused by a network problem or the client shutting down
    - client handshake fail: the client fails to handshake with TiProxy
    - auth fail: the access is denied by TiDB
    - SQL error: TiDB returns other SQL errors
    - proxy shutdown: TiProxy is shutting down
    - malformed packet: TiProxy fails to parse the MySQL packet
    - get backend fail: TiProxy fails to find an available backend for the connection
    - proxy error: other TiProxy errors
    - backend network break: fails to read from or write to the TiDB. This may be caused by a network problem or the TiDB server shutting down
    - backend handshake fail: TiProxy fails to handshake with the TiDB server
```